DEV-INTRO.md

```diff
@@ -1,59 +1,187 @@
-* General
-** all
-** utils
-** db
-* Analysis
-** modules/meta.analysis 
-** modules/uncertainty"
-** modules/data.land 
-** modules/data.atmosphere
-** modules/assim.batch 
-** modules/assim.sequential 
-** modules/priors
-* Model Interfaces
-** models/ed 
-** models/sipnet 
-** models/biocro
+If running on a linux system it is recommended to add your user to the docker group. This will prevent you from having to use `sudo` to start the docker containers, and makes sure that any file that is written to a mounted volume is owned by you. This can be done using `sudo adduser ${USER} docker`.
 
+To get started with development in docker we need to bring up the docker stack first. In the main pecan folder you will find the [docker-compose.yml](docker-compose.yml) file that can be used to bring up the pecan stack. There is also the [docker-compose.dev.yaml](docker-compose.dev.yaml) file that adds additional containers, and changes some services to make it easier for development.
 
-#### List of modules 
+By default docker-compose will use the files `docker-compose.yml` and `docker-compose.override.yml`. We will use the default `docker-compose.yml` file from PEcAn. The `docker-compose.override.yml` file can be used to configure it for your specific environment, in our case we will use it to setup the docker environment for development. To do this we copy the `docker-compose.dev.yml` file to `docker-compose.override.yml`. You can now use the command `docker-compose` to launch all the containers setup for develoopment.
 
-Installing PEcAn
-----------------
+If you in the past had loaded some of the data, but would like to start from scratch you can simply remove the `volumes` folder (and all the subfolders) and start with the "First time setup" section again.
 
-### Virtual Machine
+### First time setup
 
-* Fastest way to get started
-* see PEcAn demo ...
+The steps in this section only need to be done the fist time you start working with the stack in docker. After this is done you can skip these steps. You can find more detail about the docker commands in the [pecan documentation](https://pecanproject.github.io/pecan-documentation/master/docker-index.html).
 
-### Installing from source
+Before doing anything it is recommended to make sure you have the lastest docker images ready. You can do a `docker-compose pull` to get the latest images.
 
-#### From GitHub
+* setup .env file
+* create folders to hold the data
+* load the postgresql database
+* load some test data
+* copy all R packages (optional but recommended) 
+* setup for web folder development (optional)
 
+#### .env file
+
+You can copy the [`env.example`](docker/env.example) file as .env in your pecan folder. The variables we want to modify are:
+
+* `COMPOSE_PROJECT_NAME` set this to pecan, the prefix for all containers
+* `PECAN_VERSION` set this to develop, the docker image we start with
+
+At the end you should see the following if you run the following command `egrep -v '^(#|$)' .env`
```

Comment:
was `egrep -v '^(#|$)' .env` supposed to uncomment `COMPOSE_PROJECT_NAME` and `PECAN_VERSION` tags? because it didn't do anything for me.

Reply:
I think this is just to check that the `.env` is set up correctly? Not to change anything in the file.

Reply:
Oh I see, I misread

Reply:
I think it should include explicit instructions to uncomment those lines though, so I added that as a suggestion.